src/basis/ui/field.js

```diff
@@ -334,13 +337,45 @@
       this.setValidity();
     },
 
+    setRequired: function(value){
+      value = Boolean(resolveValue(this, this.setRequired, value, 'requiredRA_'));
+
+      if (this.init !== true || this.required !== value)
+      {
+        this.required = value;
+
+        if (value)
+          this.prependValidator(Validator.Required);
+        else
+          this.detachValidator(Validator.Required);
+      }
+    },
+    prependValidator: function(validator, validate){
```

Comment:
It would be better to add extra argument to `attachValidator()` instead of adding new method.